### ISSUE TITLE
Atom searches returns results from file types it shouldn't be searching

### ISSUE THREAD

**Issue by darylknight** [atom/atom#13611]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When I search for something in my project files using a wildcard to search through file types, such as `*.less`, Atom returns results from other file types

### Steps to Reproduce

1. Search for `px` in `*.less*
2. Atom returns `px` results in all file types such as `.html`.

**Expected behavior:** [What you expect to happen]
I only want to see results in .less files

**Actual behavior:** [What actually happens]
Atom returns results from all file types

**Reproduces how often:** [What percentage of the time does it reproduce?]
All the time

### Versions

1.13.0

**Comment by lee-dohm** [comment]
I wasn't able to reproduce this using Atom v1.13.1 on Mac OS X 10.12.3. Here's what I did:

1. Launched Atom in my local Atom repository (which has lots of Less files)
1. Created a file named `foo.txt` that contained the text `1px`
1. Opened Find in Project using <kbd>Cmd+Shift+F</kbd>
1. Entered `px` in the Find box
1. Entered `*.less` in the Filter box
1. Pressed <kbd>Enter</kbd>

**Expected and actual:** All of the results were in Less files and the result in `foo.txt` was not found

**Comment by mmcken3** [comment]
I was also not able to reproduce this using Atom v1.13.1 on Mac OS X 10.10.5. I followed the exact same steps as above to attempt to reproduce in a repository with a lot of `.less` files. Along with that I tried to recreate searching different file types such as searching through a large web app repository for `linkList` in `*.js` files.

Expected and actual: All of the results in the searches were files of `*.less` that contained `px` and files of other type with the same text were filtered out. On second attempt all files were `*.js` that contained `linkList` and other file types were filtered out.
